packages/indicators/src/types.ts

```diff
@@ -42,31 +41,30 @@ export type Indicator = {
 
 type IndicatorFields = Indicator & { id: string };
 
+type DataSourceFields = {
+  id: string;
+  code: string;
+  type: 'dataElement' | 'dataGroup';
+  service_type: 'dhis' | 'indicator' | 'tupaia' | 'weather';
+  config: Record<string, unknown>;
+};
+
+export type DataSourceType = DatabaseType<DataSourceFields>;
+
 export type IndicatorType = DatabaseType<IndicatorFields>;
 
+type DataSourceModel = DatabaseModel<DataSourceFields> & {
+  findOrDefault: DatabaseModel<DataSourceFields>['find'];
+};
+
 export interface ModelRegistry {
+  readonly dataSource: DataSourceModel;
   readonly indicator: DatabaseModel<IndicatorFields>;
 }
 
-export interface IndicatorApiInterface {
-  getAggregator: () => Aggregator;
-  buildAnalyticsForIndicators: (
-    indicators: Indicator[],
-    fetchOptions: FetchOptions,
-  ) => Promise<Analytic[]>;
-}
-
-export interface Builder {
-  (input: {
-    api: IndicatorApiInterface;
-    config: IndicatorFields['config'];
-    fetchOptions: FetchOptions;
-  }): Promise<AnalyticValue[]>;
-}
-
 export interface Aggregation {
   readonly type: string;
   readonly config?: Record<string, unknown>;
 }
 
-export type FetchOptions = Readonly<Record<string, unknown>>;
+export type FetchOptions = Readonly<{ startDate: string; endDate: string; period?: string }>;
```

Comment:
`startDate` and `endDate` are mandatory since they are required by `@tupaia/data-api`, which doesn't support `period... `period` can also be passed in since DHIS supports it, but should probably be deprecated